### PR TITLE
Understanding Cloud-Based Data Architectures

### LINKED ISSUE #6349: [Cloud Computing] Understanding cloud-based data architectures
#### Proposed title of article
Understanding cloud-based data architectures

#### Proposed article introduction
Most global enterprises are relying heavily on big data, which assist in uncovering hidden patterns. For this reason, organizations and businesses are switching to new data technologies in order to remain competitive in the market.
However, businesses operating with legacy infrastructure are finding it difficult to utilize these new technologies. This is the main reason that is making businesses migrate to cloud-based technologies.
Nowadays, cloud computing is generating billions of US dollars to businesses in revenue meaning that cloud data architecture is very important to the digital economy as it provides speed and agility and also quick access of data as well as data unification. 
In this proposed article, we will cover in detail the cloud-based data architectures and how businesses can utilize them to maximise their returns.

#### Key takeaways 
At the end of the article, the learner will be able to understand:
- the concept of cloud data architectures.
- the core components of cloud computing.
- the factors considered when building a stable cloud data architecture.

#### Article quality
The proposed article will expound on cloud data architectures, highlighting the cloud computing architectures in relation to the data. The businesses can then utilize the information provided to build their stable data architecture that is cloud-based.

#### References
n/a

### PR DESCRIPTION
**Pull request by verah-tech**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #6349

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/understanding-cloud-based-data-architectures/index.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+The article covers [cloud-based data architectures](https://www.snowflake.com/trending/cloud-data-architectures) and how businesses can use them to maximize their returns.
+
+### The concept of cloud data architectures
+Most applications use the traditional database architecture to integrate similar [structured](https://www.tibco.com/reference-center/what-is-structured-data) data types. However, the complexity of the data involved may not favour the local parameters. Besides, the on-premises storage architecture is usually expensive to build and maintain. It is impractical to maintain the speed and flexibility of modern datasets in this era of big data.
+
+Thus, cloud-based data warehouse architecture is favourable to curb the above challenges. It makes efficient use of the data warehouse resources. Cloud data architecture contains the rules, models, and policies that handle data. It dictates how data is gathered and stored in the cloud platforms in an organization.
+
+Below are the benefits of cloud data architectures over the classic architectures:
```
Comment (mbaabuones) [inline content/articles/understanding-cloud-based-data-architectures/index.md:13]: You can create a new section 'Benefits of cloud data architectures'. Here, explain the specific benefits in-depth.

File: content/articles/understanding-cloud-based-data-architectures/index.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+- Fast data access rates.
+- Low computing costs.
+- Supports future technological improvements.
+- Access to real-time analytics and data processing.
+
+### The core components of cloud computing
+Most cloud computing architecture consists of two main components: frontend and backend. The [frontend architecture](https://www.oreilly.com/library/view/frontend-architecture-for/9781491926772/ch01.html) contains every component that the end-user can access. This ranges from user interface components to the client infrastructure. The [backend](https://www.codecademy.com/article/back-end-architecture) handles data storage, server, operating system, security, and other backend procedures. It also handles security protocols that help to keep the cloud services secure.
```
Comment (mbaabuones) [inline content/articles/understanding-cloud-based-data-architectures/index.md:20]: Here you can be innovative by creating a list of the two components and explaining them one by one. eg.
- **Front end architecture:** This contains every component.......
- **Back end architecture:** This handles data storage, ....

File: content/articles/understanding-cloud-based-data-architectures/index.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+For an organization to migrate to a cloud computing solution, it must be connected to the network. There are different ways an organization can migrate its data to the cloud, as stated below:
```
Comment (mbaabuones) [inline content/articles/understanding-cloud-based-data-architectures/index.md:22]: This information does not directly correspond to the title of the section. Here, you can create a new section called 'Migrating data to the cloud'.

File: content/articles/understanding-cloud-based-data-architectures/index.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+- **Migration of existing data and applications:** It requires programming effort, which is costly. The method is also known as the [lift and shift strategy](https://www.ibm.com/cloud/learn/lift-and-shift).
+- **Implementing hybrid data infrastructure:** It is the most popular option organizations consider. It involves expanding the on-premises data infrastructure into a hybrid infrastructure. It offers better security and allows users to enjoy the two data infrastructures.
+
+### Factors considered when building a stable cloud data architecture
+Reliable data architecture takes time to build. All necessary components must be well integrated to form a seamless system. Data architects have to consider some practices when designing cloud data architecture.
+
+Furthermore, there is no all-in-one solution for developing secure and efficient data architecture. The time required can vary depending on the organization's business needs. Below are some factors to consider when developing a cloud data infrastructure:
+
+#### Understanding the cloud architecture
```
Comment (mbaabuones) [inline content/articles/understanding-cloud-based-data-architectures/index.md:33]: The cloud model

File: content/articles/understanding-cloud-based-data-architectures/index.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+Organizations should understand various cloud services that different vendors in the market offer. Cloud architectures belong into four different categories as below:
+- **Private cloud:** In this category, the cloud services are used by one organization. This includes the employees, customers or stakeholders.
+- **Public cloud:** The cloud services here are open to the public. The provider owns the cloud infrastructure here.
+- **Hybrid cloud:** Here, the public and private cloud services are combined to meet the organizations' specific needs.
+- **Multi-cloud:** The above three cloud services are patched and combined to offer optimal computing capabilities at a minimal cost.
+
+Organizations should choose the cloud model that fits their unique needs and preferences.
+
+#### Understand different cloud storage options
```
Comment (mbaabuones) [inline content/articles/understanding-cloud-based-data-architectures/index.md:42]: Cloud storage options

**Review by mbaabuones — CHANGES_REQUESTED** [review]
@verah-tech Thank you for submitting your article. Your article is informative and easy to understand. After running your article through Grammarly.com, I have found some minor grammatical errors. Run it through Grammarly.com to identify and fix these errors. Please work on the minor suggested changes and ping me when you are done. Remember to proofread your article after doing the revision.

**Comment by verah-tech** [comment]
Hi @mbaabuones, am grateful for taking the time to review my article. The changes requested have been made and noted. Kindly proceed with the review.
Thank you.

**Comment by lalith1403** [comment]
plagiarism check done ✔️

**Review by mbaabuones — APPROVED** [review]

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**